ratelimit_test.go

```diff
@@ -13,6 +13,82 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type runner interface {
+	// startTaking tries to Take() on passed in limiters in a loop/goroutine.
+	startTaking(rls ...ratelimit.Limiter)
+	// assertCountAt asserts the limiters have Taken() a number of times at the given time.
+	// It's a thin wrapper around afterFunc to reduce boilerplate code.
+	assertCountAt(d time.Duration, count int)
+	// getClock returns the test clock.
+	getClock() ratelimit.Clock
+	// afterFunc executes a func at a given time.
+	afterFunc(d time.Duration, fn func())
+}
+
+type runnerImpl struct {
+	t *testing.T
+
+	clock *clock.Mock
+	count atomic.Int32
+	// maxDuration is the time we need to move into the future for a test.
+	// It's populated automatically based on assertCountAt/afterFunc.
+	maxDuration time.Duration
+	doneCh      chan struct{}
+	wg          sync.WaitGroup
+}
+
+func runTest(t *testing.T, fn func(runner)) {
+	r := runnerImpl{
+		t:      t,
+		clock:  clock.NewMock(),
+		doneCh: make(chan struct{}),
```

Comment:
nit: `done` is probably good enough?

Reply:
hah, I thought this style was a recommendation? I guess I was wrong :)

Reply:
Oh, sorry, I thought I had pushed/changed this, but apparently didn't.

Will fix in a follow up.